enhancements/installer/aws-cross-account-dns-zone.md

```diff
@@ -0,0 +1,235 @@
+The name of the field is discussed further in open questions.
+
+#### OpenShift API
+
+```go
+type AWSPlatformStatus struct {
+	// ...existing fields...
+
+  // privateHostedZoneRole contains the ARN of a role that should be assumed when performing
+  // operations on the cluster's private hosted zone specified in the cluster DNS config.
+  // When left empty, no role should be assumed.
+  // +kubebuilder:validation:Pattern:=`^arn:(aws|aws-cn|aws-us-gov):iam:[a-z0-9-]+:[0-9]{12}:role\/.*$`
+  // +optional
+  // PrivateHostedZoneRole string `json:"privateHostedZoneRole, omitempty"`
+}
+```
+
+#### Install Config
+
+`platform.aws.hostedZoneRole` would be added to complement the existing `platform.aws.hostedZone` field.
+
+An example of `platform` from `install-config.yaml`, specifying subnets of the shared VPC,
+a hosted zone ID, and a role to assume:
+
+```yaml
+platform:
+  aws:
+    region: us-east-2
+    subnets:
+    - subnet-0ef71407239a73558
+    - subnet-00e576c91f8578661
+    - subnet-041af6e1bbf29c2ef
+    - subnet-037ab7dc0cd396721
+    - subnet-0adda52a1007fc74f
+    - subnet-09bd88678851acab6
+    hostedZone: Z0743278N5ELJPLYA3H6
+    hostedZoneRole: arn:aws:iam::<account-a>:role/<role-name>
+```
+Please see Open Questions for further discussion of the install config.
+
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+ With the implementation of this API, the installer and cluster components should follow a
+ relatively straightforward pattern: if the role field is not empty,
+ call [`AssumeRole`](https://docs.aws.amazon.com/sdk-for-go/api/aws/credentials/stscreds/)
+ when performing operations on the private hosted zone.
+
+### Risks and Mitigations
+
+n/a
+
+### Drawbacks
+
+n/a
+
+## Design Details
+
+### Open Questions [optional]
+
+1. The suggested field name for the API is clunky, but has the advantage of being least ambiguous.
+Some other possible candidates would be:
+- sharedVPCRole
+- privateDNSRole
+- privateZoneRole
+- privateDNSZoneRole
+- dnsRole
+- dnsZoneRole
+- privateHostedZoneRole
+2. In the install config, whould we consider deprecating `aws.platform.hostedZone` in order to
+create a struct that can nest both the `id` and `role` fields? Using the same example as above,
+the install config would look like:
+
+```yaml
+platform:
+  aws:
+    region: us-east-2
+    subnets:
+    - subnet-0ef71407239a73558
+    - subnet-00e576c91f8578661
+    - subnet-041af6e1bbf29c2ef
+    - subnet-037ab7dc0cd396721
+    - subnet-0adda52a1007fc74f
+    - subnet-09bd88678851acab6
+    hostedZone: #deprecated
+    privateHostedZone:
+      id: Z0743278N5ELJPLYA3H6
+      role: arn:aws:iam::<account-a>:role/<role-name>
+```
+
+3. Cluster admins should be able to update the role in the API, which is why I included it in the
+spec. Is that the correct place for it? I frequently get confused about how this should work.
```

Comment:
Right now, the EP shows the new field in `AWSPlatformStatus`, not `AWSPlatformSpec`.  

Is there a use-case for modifying the role day-2?  

If you do make the role mutable, would that also make it possible to upgrade a cluster and then change a running cluster from using a private hosted zone in account-b (with `role` empty, not using a shared VPC) to using a private hosted zone in account-a (with `role` nonempty to specify a shared VPC) or vice versa?  

Also, if you make the role mutable, maybe it should be kept with the DNS zone configuration (which is in `dnses.config.openshift.io/cluster`), so that the role and zone id can be changed in a single update to one object (as opposed to requiring two updates, to `dnses.config.openshift.io/cluster` for the zone id and to `infrastructures.config.openshift.io/cluster` for the role)?

Reply:
I have updated the enhancement to prefer updating the dns config. I'm going to leave this open for discussion of day 2.